### ISSUE TITLE
Error in transaction creation based on Kafka record

### ISSUE THREAD

**Issue by pruthvi6767** [elastic/apm-agent-java#1840]
Kibana version:
7.11.2
Elasticsearch version:
7.11.2
APM Server version:
7.11.2
APM Agent language and version:
java - 1.22.0

Original install method (e.g. download page, yum, deb, from source, etc.) and version:
java client jar

Description of the problem including expected versus actual behavior. Please include screenshots

Our Application logs show

_ERROR co.elastic.apm.agent.kafka.helper.ConsumerRecordsIteratorWrapper - Error in transaction creation based on Kafka record_

Our insights are around a message headers key/value having a null value and apm failing to capture such messages .

Provide logs and/or server output (if relevant):

 ```
 at java.lang.Thread.run(Thread.java:834) [?:?]
    	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
    	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
    	at org.springframework.kafka.listener.KafkaMessageListenerContainer$ListenerConsumer.run(KafkaMessageListenerContainer.java:1073) [spring-kafka-2.6.4.jar!/:2.6.4]
    	at org.springframework.kafka.listener.KafkaMessageListenerContainer$ListenerConsumer.pollAndInvoke(KafkaMessageListenerContainer.java:1176) [spring-kafka-2.6.4.jar!/:2.6.4]
    	at org.springframework.kafka.listener.KafkaMessageListenerContainer$ListenerConsumer.invokeListener(KafkaMessageListenerContainer.java:1529) [spring-kafka-2.6.4.jar!/:2.6.4]
    	at org.springframework.kafka.listener.KafkaMessageListenerContainer$ListenerConsumer.invokeRecordListener(KafkaMessageListenerContainer.java:1810) [spring-kafka-2.6.4.jar!/:2.6.4]
    	at org.springframework.kafka.listener.KafkaMessageListenerContainer$ListenerConsumer.doInvokeWithRecords(KafkaMessageListenerContainer.java:1911) [spring-kafka-2.6.4.jar!/:2.6.4]
    	at co.elastic.apm.agent.kafka.helper.ConsumerRecordsIteratorWrapper.next(ConsumerRecordsIteratorWrapper.java:42) [?:?]
    	at co.elastic.apm.agent.kafka.helper.ConsumerRecordsIteratorWrapper.next(ConsumerRecordsIteratorWrapper.java:100) [?:?]
    	at co.elastic.apm.agent.impl.context.Message.addHeader(Message.java:127) ~[?:?]
    	at co.elastic.apm.agent.impl.context.Headers.add(Headers.java:47) ~[?:?]
    	at co.elastic.apm.agent.util.BinaryHeaderMap.add(BinaryHeaderMap.java:76) ~[?:?]
    	at co.elastic.apm.agent.util.IOUtils.decodeUtf8Bytes(IOUtils.java:130) ~[?:?]
    java.lang.NullPointerException: null

```

**Comment by SylvainJuge** [comment]
For reference, this is also related to the [following forum discussion](https://discuss.elastic.co/t/error-in-transaction-creation-based-on-kafka-record/274927).

**Comment by SylvainJuge** [comment]
Hi @pruthvi6767 , I've opened a PR that fixes the NPE issue, could you try the [following snapshot](https://apm-ci.elastic.co/job/apm-agent-java/job/apm-agent-java-mbp/job/PR-1842/1/artifact/src/github.com/elastic/apm-agent-java/elastic-apm-agent/target/elastic-apm-agent-1.24.1-SNAPSHOT.jar) in order to be sure it properly fixes the issue you experienced ?

**Comment by pruthvi6767** [comment]
Hello @SylvainJuge  Please let us know when this fix would be released?

**Comment by SylvainJuge** [comment]
Hi @pruthvi6767 , it will be included in the next release (`1.25.0`), thus make sure to subscribe to project notifications to ensure proper notification when it's released.

In the mean time, you can use the [snapshot version](https://github.com/elastic/apm-agent-java/#snapshots) to make sure it properly covers your case.
